File: plugins/scandit-sdk/evals/label-capture-capacitor/fixtures/before-8.4-validation-flow.js
import { Camera, DataCaptureContext, DataCaptureView, FrameSourceState, ScanditCaptureCorePlugin } from 'scandit-capacitor-datacapture-core';
import { Symbology } from 'scandit-capacitor-datacapture-barcode';
import {
  CustomBarcode,
  ExpiryDateText,
  LabelCapture,
  LabelCaptureSettings,
  LabelCaptureValidationFlowOverlay,
  LabelDefinition,
} from 'scandit-capacitor-datacapture-label';

let labelCapture;

async function runApp() {
  await ScanditCaptureCorePlugin.initializePlugins();
  const context = DataCaptureContext.initialize('-- ENTER YOUR SCANDIT LICENSE KEY HERE --');

  const camera = Camera.withSettings(LabelCapture.createRecommendedCameraSettings());
  context.setFrameSource(camera);

  const barcode = CustomBarcode.initWithNameAndSymbologies('Barcode', [
    Symbology.EAN13UPCA,
    Symbology.Code128,
  ]);
  barcode.optional = false;

  const expiry = new ExpiryDateText('Expiry Date');
  expiry.optional = false;

  const labelDefinition = new LabelDefinition('Perishable Product');
  labelDefinition.fields = [barcode, expiry];

  const settings = LabelCaptureSettings.settingsFromLabelDefinitions([labelDefinition], {});
  labelCapture = new LabelCapture(settings);
  context.setMode(labelCapture);

  const view = DataCaptureView.forContext(context);
  view.connectToElement(document.getElementById('data-capture-view'));

  const validationFlowOverlay = new LabelCaptureValidationFlowOverlay(labelCapture);
  validationFlowOverlay.listener = {
    didCaptureLabelWithFields(fields) {
      labelCapture.isEnabled = false;
      view.webViewContentOnTop = true;
      console.log(fields);
    },
    didSubmitManualInputForField(field, oldValue, newValue) {
      console.log('Manual input', field.name, oldValue, '->', newValue);
    },
  };
  await view.addOverlay(validationFlowOverlay);

  view.webViewContentOnTop = false;
  await camera.switchToDesiredState(FrameSourceState.On);
  labelCapture.isEnabled = true;
}

runApp();

File: plugins/scandit-sdk/evals/label-capture-flutter/fixtures/before-8.4-validation-flow.dart
import 'package:flutter/material.dart';
import 'package:scandit_flutter_datacapture_core/scandit_flutter_datacapture_core.dart';
import 'package:scandit_flutter_datacapture_barcode/scandit_flutter_datacapture_barcode.dart';
import 'package:scandit_flutter_datacapture_label/scandit_flutter_datacapture_label.dart';

const licenseKey = '-- ENTER YOUR SCANDIT LICENSE KEY HERE --';

Future<void> main() async {
  WidgetsFlutterBinding.ensureInitialized();
  await ScanditFlutterDataCaptureBarcode.initialize();
  await ScanditFlutterDataCaptureLabel.initialize();
  DataCaptureContext.initialize(licenseKey);
  runApp(const MaterialApp(home: ScanScreen()));
}

class ScanScreen extends StatefulWidget {
  const ScanScreen({super.key});
  @override
  State<ScanScreen> createState() => _ScanScreenState();
}

class _ScanScreenState extends State<ScanScreen>
    implements LabelCaptureValidationFlowExtendedListener {
  late final DataCaptureContext _context;
  late final Camera? _camera;
  late final LabelCapture _labelCapture;
  late final DataCaptureView _view;
  late final LabelCaptureValidationFlowOverlay _overlay;

  @override
  void initState() {
    super.initState();
    _context = DataCaptureContext.sharedInstance;
    _camera = Camera.defaultCamera;
    _camera?.applySettings(LabelCapture.createRecommendedCameraSettings());
    _context.setFrameSource(_camera);

    final barcode = CustomBarcodeBuilder()
        .setSymbologies({Symbology.ean13Upca, Symbology.code128})
        .isOptional(false)
        .build('Barcode');
    final expiry = ExpiryDateTextBuilder().isOptional(false).build('Expiry Date');
    final definition = LabelDefinitionBuilder()
        .addCustomBarcode(barcode)
        .addExpiryDateText(expiry)
        .build('Perishable Product');

    _labelCapture = LabelCapture.forContext(_context, LabelCaptureSettings([definition]));
    _view = DataCaptureView.forContext(_context);
    _overlay = LabelCaptureValidationFlowOverlay.withLabelCaptureForView(_labelCapture, _view);
    _overlay.listener = this;

    _camera?.switchToDesiredState(FrameSourceState.on);
    _labelCapture.isEnabled = true;
  }

  @override
  void didCaptureLabelWithFields(List<LabelField> fields) {
    final body = fields
        .map((f) => '${f.name}: ${f.barcode?.data ?? f.text ?? 'N/A'}')
        .join('\n');
    showDialog(
      context: context,
      builder: (_) => AlertDialog(
        title: const Text('LABEL CAPTURED'),
        content: Text(body),
        actions: [
          TextButton(onPressed: () => Navigator.pop(context), child: const Text('OK')),
        ],
      ),
    );
  }

  @override
  void didSubmitManualInputForField(LabelField field, String? oldValue, String newValue) {
    // User manually corrected a field value.
  }

  @override
  void dispose() {
    _overlay.listener = null;
    super.dispose();
  }

  @override
  Widget build(BuildContext context) => Scaffold(body: _view);
}
